contracts/pol/CarbonPOL.sol

```diff
@@ -0,0 +1,303 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.19;
+
+import { ReentrancyGuardUpgradeable } from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+
+import { IVersioned } from "../utility/interfaces/IVersioned.sol";
+import { ICarbonPOL } from "./interfaces/ICarbonPOL.sol";
+import { Upgradeable } from "../utility/Upgradeable.sol";
+import { Token } from "../token/Token.sol";
+import { Utils } from "../utility/Utils.sol";
+import { MathEx } from "../utility/MathEx.sol";
+import { ExpDecayMath } from "../utility/ExpDecayMath.sol";
+import { MAX_GAP, PPM_RESOLUTION } from "../utility/Constants.sol";
+
+/**
+ * @notice CarbonPOL contract
+ */
+contract CarbonPOL is ICarbonPOL, Upgradeable, ReentrancyGuardUpgradeable, Utils {
+    using Address for address payable;
+
+    // initial starting price multiplier for the dutch auction
+    uint32 private _marketPriceMultiply;
+
+    // time until the price gets back to market price
+    uint32 private _priceDecayHalfLife;
+
+    // token to trading start time mapping
+    mapping(Token token => uint32 tradingStartTime) private _tradingStartTimes;
+
+    // token to initial price mapping
+    mapping(Token token => Price initialPrice) private _initialPrice;
+
+    // upgrade forward-compatibility storage gap
+    uint256[MAX_GAP - 3] private __gap;
+
+    /**
+     * @dev used to initialize the implementation
+     */
+    constructor() {
+        // initialize implementation
+        initialize();
+    }
+
+    /**
+     * @dev fully initializes the contract and its parents
+     */
+    function initialize() public initializer {
+        __CarbonPOL_init();
+    }
+
+    // solhint-disable func-name-mixedcase
+
+    /**
+     * @dev initializes the contract and its parents
+     */
+    function __CarbonPOL_init() internal onlyInitializing {
+        __Upgradeable_init();
+        __ReentrancyGuard_init();
+
+        __CarbonPOL_init_unchained();
+    }
+
+    /**
+     * @dev performs contract-specific initialization
+     */
+    function __CarbonPOL_init_unchained() internal onlyInitializing {
+        // set market price multiplier to 2x
+        _setMarketPriceMultiply(2);
+        // set price decay half-life to 10 days
+        _setPriceDecayHalfLife(10 days);
+    }
+
+    /**
+     * @dev authorize the contract to receive the native token
+     */
+    receive() external payable {}
```

Comment:
Why is this needed? The contract shouldn't receive ETH explicitly (since it can't really offer it for trading) and traded ETH is receiving with an explicit `payable` function anyway.

Reply:
I see. Perhaps worth adding a suitable comment so that it doesn't appear as a mistake. Resolving in any case.